alacritty_terminal/src/selection.rs

```diff
@@ -272,12 +272,31 @@ impl Selection {
 
     fn span_lines<T>(term: &T, mut start: Point<isize>, mut end: Point<isize>) -> Option<Span>
     where
-        T: Dimensions,
+        T: Search + Dimensions,
     {
-        start.col = term.dimensions().col - 1;
-        end.col = Column(0);
+        let start_span: Span = term.search_wrapline(start.into());
+        let end_span: Span = term.search_wrapline(end.into());
+
+        if start_span.end.line == end_span.end.line {
+            start = start_span.start.into();
+            end = end_span.end.into();
+            start.line += 1;
+        } else if start_span.start.line > end_span.end.line {
+            start = start_span.start.into();
+            end = end_span.end.into();
+            end.line -= 1;
+        } else {
+            start = start_span.end.into();
+            end = end_span.start.into();
+            end.line += 1;
+        }
```

Comment:
All of this code shouldn't be necessary if the search left/right methods would just return the point after expanding it all the way to the left/right. The same way the semantic search functions.